macros/g4simulations/G4_BlackHole.C

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "GlobalVariables.C"
+
+#include <g4detectors/PHG4CylinderSubsystem.h>
+#include <g4main/PHG4Reco.h>
+
+R__LOAD_LIBRARY(libg4detectors.so)
+
+
+void BlackHoleInit() {}
+
+void BlackHole(PHG4Reco *g4Reco, double radius)
+{
+  // swallow all particles coming out of our detector
+  if (radius < BlackHoleGeometry::max_radius)
+  {
+    radius = BlackHoleGeometry::max_radius;
+  }
+  double blackhole_length = (BlackHoleGeometry::max_z - BlackHoleGeometry::min_z)+2*BlackHoleGeometry::gap;
+  double blackhole_zpos = BlackHoleGeometry::min_z - BlackHoleGeometry::gap + blackhole_length/2.;
+  double blackhole_radius = radius + BlackHoleGeometry::gap; // make the black hole slightly larger than the radius
+  PHG4CylinderSubsystem *blackhole = new PHG4CylinderSubsystem("BH", 1);
+  blackhole->set_double_param("radius",blackhole_radius); 
+  blackhole->set_double_param("length", blackhole_length); // make it cover the world in length
+  blackhole->set_double_param("place_z",blackhole_zpos);
+  blackhole->set_double_param("thickness",BlackHoleGeometry::gap/2.); // it needs some thickness
+  if (BlackHoleGeometry::visible)
+  {
+    blackhole->set_color(1,0,0,0.7);
+  }
+  blackhole->BlackHole();
+  blackhole->SetActive(); // always see what leaks out
+  blackhole->OverlapCheck(true);
```

Comment:
Less verbose in default running with following?
```
blackhole->OverlapCheck(overlap);
```
Similar for the next three instances